### DISCUSSION TITLE
How to convert a token's type

### DISCUSSION THREAD

**Discussion by spifory** [markedjs/marked#3476] [Q&A] [answered]
During `walkTokens`, with an `image` type, I'd like to do a try/catch that tries `fetch`ing the href and if it fails to do so, then it should change the entire thing from `image` to `type`. So for instance:
```ts
	try {
		// ![](https://invalid-image.com)
		const image = await fetch(token.href)
	} except {
		token.text = 'link'
		token.type = 'link' // Something like this, however it doesn't seen to work when I do this
		// [link](https://invalid-image.com)
	}
```

**Comment by UziTech** [comment, marked as answer]
That will work but link has sub tokens in `token.tokens` that the parser uses to render the text.

So it will have to be something like

```
  try {
		// ![](https://invalid-image.com)
		const image = await fetch(token.href)
	} except {
		token.text = 'link'
		token.type = 'link'
    token.tokens = lexer.inlineTokens(token.text); // This will tell the parser what to output
	}
```

**Reply by spifory** [reply]
Ah okay I see. Yeah works out pretty well, but I'm curious if there's a way for it to this without marked turning the token into HTML code? Currently it returns this as `<a href="http://invalid.url">ee</a>` when I would like for it to be returned as `[ee](http://invalid.url)`. 

In case this is useful, this try/catch is under my `walkTokens` that gets passed into a `await parseInline(..., { walkTokens })`

**Reply by UziTech** [reply]
You want marked to output markdown?

You will have to update the [renderers](https://marked.js.org/using_pro#renderer). You can just return the `token.raw` property for most of them.

**Reply by spifory** [reply]
Awesome, thanks for the help!
